tt_eager/tt_dnn/op_library/untilize/multi_core/untilize_op_multi_core.cpp

```diff
@@ -209,21 +211,10 @@ operation::ProgramWithCallbacks untilize_multi_core(
     }
 
     // 1D distribution of blocks across all cores
-    uint32_t ncores_full = ncores;
-    auto full_cores = all_cores;
-    if (nblocks_per_core_cliff > 0 && nblocks_per_core_cliff < nblocks_per_core) {
-        // unequal case with cliff
-        ncores_full -= 1;
-        full_cores = core_range;
-    }
     uint32_t tile_start_id = 0;
     uint32_t row_start_id = 0;
-    auto cores = grid_to_cores(ncores_x * ncores_y, ncores_x, ncores_y, row_major);
-    for (uint32_t i = 0; i < cores.size(); i++) {
-        CoreCoord core = cores[i];
-        if (!full_cores.core_coord_in_core_ranges(core)) {
-            continue;
-        }
+
+    for (const auto& core : iterate(core_range, true)) {
```

Comment:
`auto enumerate(Iterable&& iterable, std::size_t start = 0) {`

but `iterate(core_range, true))`

What does `true` means here?

Reply:
Not possible to just `for (const auto& core : core_range) {`?

Reply:
`true` for the direction of the iteration (row-first or col-first), otherwise I could have iterated directly over the range.